q2_annotate/busco/busco.py

```diff
@@ -140,6 +154,24 @@ def _evaluate_busco(
     common_args = _process_common_input_params(
         processing_func=_parse_busco_params, params=kwargs
     )
+    if isinstance(mags, MultiMAGSequencesDirFmt):
+        busco_results = _busco_helper(mags, common_args, additional_metrics)
+        busco_results["unbinned_contigs"] = pd.NA
+        busco_results["unbinned_contigs_count"] = pd.NA
+        for unbinned_id, unbinned_path in unbinned_contigs.sample_dict().items():
+            binned_dir = mags.path / unbinned_id
+            binned_fasta_paths = _get_fasta_files_from_dir(binned_dir)
+            percentage, count = _calculate_unbinned_percentage(
```

Comment:
This is technically fine the way it's written but I think it would be a bit neater if, instead of updating the results on every iteration, you collected all the results into a pd.DataFrame and then just merged this DataFrame witht the original results DF at the very end. This way you need to temper with the results table only once and can take care of the data formats in all the columns in the intermediate DataFrame (which most likely will just happen automatically).